plugin-unity-agent/src/main/kotlin/jetbrains/buildServer/unity/PEProductVersionDetector.kt

```diff
@@ -0,0 +1,76 @@
+package jetbrains.buildServer.unity
+
+import com.intellij.openapi.diagnostic.Logger
+import com.vdurmont.semver4j.Semver
+import jetbrains.buildServer.util.PEReader.PEUtil
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import java.io.BufferedReader
+import java.io.File
+import java.nio.file.Paths
+import java.util.concurrent.TimeUnit
+import kotlin.io.path.exists
+
+class PEProductVersionDetector {
+    fun detect(executable: File): Semver? {
+        val version = PEUtil.getProductVersion(executable)
+
+        if (version != null) {
+            return Semver("${version.p1}.${version.p2}.${version.p3}", Semver.SemverType.LOOSE)
+        }
+
+        LOG.debug("unable to detect version via PEUtil, falling back to detection tool")
+
+        val detectorPath = Paths.get("../plugins/teamcity-unity-agent/tools/PeProductVersionDetector.exe")
```

Comment:
Maybe it will be better to extract a path and executable name to different variables and put in some place where we could reach it easily from other places, e.g. from tests. What do you think about it?